### PR TITLE
listview fix https://github.com/EnsembleUI/ensemble/issues/247

### PR DESCRIPTION
**Pull request by Moeez1999**

### CODE REVIEW

**Comment by kmahmood74** [comment]
@Moeez1999 can you please make a video of @gaurikalp 's use case working fine with your fix and upload it here please. Thanks

**Comment by Moeez1999** [comment]
Sure i will

**Comment by Moeez1999** [comment]
https://user-images.githubusercontent.com/55261847/220431508-853fb82a-856f-4219-b8ff-9ae108bf5a5c.mp4

**Comment by kmahmood74** [comment]
can you actually scroll in your video for the ITI courses. Scrolling experience is critical too right? Please update your video with actually scrolling through the 950+ records and scrolling up and down

File: lib/layout/templated.dart [hunk 1]
```diff
@@ -63,6 +63,22 @@ mixin TemplatedWidgetState<W extends StatefulWidget> on State<W> {
     }
     return widgets;
   }
+  // ------------------------ Render the listview children ----------------------------------
+  DataScopeWidget? buildWidgetsForListView(BuildContext context, List dataList, ItemTemplate itemTemplate,int itemIndex) {
```
Comment (vusters) [inline lib/layout/templated.dart:67]: so this fixes #1, and shrinkWrap fixes # 2?

File: lib/layout/templated.dart [hunk 2]
```diff
@@ -63,6 +63,22 @@ mixin TemplatedWidgetState<W extends StatefulWidget> on State<W> {
...
+    DataScopeWidget? widgets;
+    ScopeManager? parentScope = DataScopeWidget.getScope(context);
```
Comment (vusters) [inline lib/layout/templated.dart:69]: Try not to copy and paste code. Can you create a function for creating individual widget inside templated.dart, and have both places use this block?

File: lib/layout/templated.dart [hunk 3]
```diff
@@ -63,6 +63,22 @@ mixin TemplatedWidgetState<W extends StatefulWidget> on State<W> {
...
+    if (parentScope != null) {
+      
+        // create a new scope for each item template
+        ScopeManager templatedScope = parentScope.createChildScope();
+        templatedScope.dataContext.addDataContextById(itemTemplate.name, dataList.elementAt(itemIndex));
+
+        Widget templatedWidget = templatedScope.buildWidgetFromDefinition(itemTemplate.template);
+
+        widgets = DataScopeWidget(scopeManager: templatedScope, child: templatedWidget);
```
Comment (vusters) [inline lib/layout/templated.dart:78]: name it singular `widget`

**Comment by vusters** [comment]
This is super nice. Nice work @Moeez1999 and @snehmehta

**Comment by kmahmood74** [comment]
@Moeez1999 @snehmehta nice work! @Moeez1999 please go ahead and merge the PR as it has been approved.

**Pull request merged**